### PR TITLE
Review RC branch changes to develop

### PR DESCRIPTION
**Pull request by rhornung67**
# Summary

- This PR is to review v2024.07.0 release candidate branch changes to develop. 

**Please review this PR, but approve https://github.com/LLNL/RAJA/pull/1696 which will be merged into main for the release.**

### CODE REVIEW

File: RELEASE_NOTES.md [hunk 1]
```diff
@@ -20,6 +20,58 @@ Notable changes include:
...
+          to use atomic loads instead of relying on volatile.
+        * Re-implemented atomic loads in terms of builtin atomics for CUDA
+          and HIP so that the generic compare and swap functions can use it.
+        * Removes volatile qualifier in atomic function signatures.
+        * Use cuda::atomic_ref in newer versions of CUDA to back 
+          atomicLoad/atomicStore.
+        * Use atomicAdd as a fallback for atomicSub in CUDA.
+        * Removed checks where __CUDA_ARCH__ is less than 350 since RAJA 
+          requires that as the minimum supported architecture (CMake check).
+     * Fix issue related to the fact that only C version of memcpy is supported
```
Comment (adayton1) [inline RELEASE_NOTES.md:61]: This was part of the churn for atomics, so I'm not sure it needs to be mentioned (it wasn't an outstanding bug from the last release).

**Review by adayton1 — APPROVED** [review]
LGTM

File: RELEASE_NOTES.md (hunk 1 above)
Comment (rhornung67) [reply RELEASE_NOTES.md:61]: The cuda arch check?

Comment (adayton1) [reply RELEASE_NOTES.md:61]: No, the C version of memcpy

Comment (rhornung67) [reply RELEASE_NOTES.md:61]: Gotcha. I will remove from release notes.

**Pull request closed without merging**